homeassistant/components/mpd/media_player.py

```diff
@@ -106,37 +108,38 @@ def __init__(self, server, port, password, name):
         self._muted_volume = 0
         self._media_position_updated_at = None
         self._media_position = None
+        self._commands = None
 
         # set up MPD client
-        self._client = mpd.MPDClient()
+        self._client = MPDClient()
         self._client.timeout = 30
         self._client.idletimeout = None
 
-    def _connect(self):
+    async def _connect(self):
         """Connect to MPD."""
         try:
-            self._client.connect(self.server, self.port)
+            await self._client.connect(self.server, self.port)
 
             if self.password is not None:
-                self._client.password(self.password)
+                await self._client.password(self.password)
         except mpd.ConnectionError:
             return
 
         self._is_connected = True
 
-    def _disconnect(self):
+    async def _disconnect(self):
         """Disconnect from MPD."""
         try:
-            self._client.disconnect()
-        except mpd.ConnectionError:
+            await self._client.disconnect()
+        except (mpd.ConnectionError, TypeError):
```

Comment:
Why do we catch TypeError?

Reply:
Looks like that error was a consequence of awaiting disconnect incorrectly.

```suggestion
        except mpd.ConnectionError:
```

Reply:
Possible, I'll roll out the updated version and we'll see how it goes.